### DISCUSSION TITLE
string + email + nullish fails when empty string

### DISCUSSION THREAD

**Discussion by mmahalwy** [colinhacks/zod#994] [Q&A]
```ts
const emailSchema = z.string().email().nullish();

emailSchema.parse("")
```

fails with invalid email. This should not be the case due to `nullish` no?

**Comment by JacobWeisenburger** [comment]
An empty string is not considered a valid email based on `emailRegex`. See source code below:

https://github.com/colinhacks/zod/blob/004eefea58827bcf122c0517268cd11a540d207b/src/types.ts#L443-L445

https://github.com/colinhacks/zod/blob/004eefea58827bcf122c0517268cd11a540d207b/src/types.ts#L487-L495

---
Nullish doesn't affect this, because `"" !== null`

**Reply by mmahalwy** [reply]
@JacobWeisenburger makes sense. I feel like either: 1. we have something that allows empty string for `string()` or 2. allow `nullish` to also cover empty string. thoughts @colinhacks ?

**Reply by michaelhays** [reply]
As a workaround, you can use [preprocess](https://zod.dev/?id=preprocess) to convert the empty string to `undefined`:

```ts
const emailSchema = z.preprocess(
  (value) => (value === '' ? undefined : value),
  z.string().email().nullish(),
)
```
